### PR TITLE
[WIP] Adobe Stock SDK: stub for the SDK

### LINKED ISSUE #593: Test Module - Adobe Stock SDK Stub
Develop a Test Module - Adobe Stock SDK Stub for running tests without credentials and network connection.

The module should add a preference for "\Magento\AdobeStockClient\Model\ConnectionFactory" and to the factory returning the extended and stubbed `\AdobeStock\Api\Client\AdobeStock` proving mock responses.

The module should be placed inside `dev/test/...` directory of `adobe-stock-integration` repo

Examples of existing test modules in Magento:

- `dev/tests/integration/_files/Magento`
- `dev/tests/api-functional/_files/Magento`

### PR DESCRIPTION
**Pull request by coderimus**
### Description (*)
This PR provides a stub module for the Adobe SDK and Adobe Stock Client module. Work is in progress. A description will be added when implementation will be approved and ready to be merged.

### Fixed Issues (if relevant)
#593 

### Manual testing scenarios (*)
N/A

### CODE REVIEW

**Comment by coderimus** [comment]
@sivaschenko I have an idea: what if we will add this stub under the `AdobeStockClietn\Test\_files` location? I think with this we will be able to use it for integration tests if needed and Web API. What do you think?

**Comment by coderimus** [comment]
@sivaschenko according to yesterday's discussion: I think I got your idea about the preference. I think I can add this Client class to the _files folder at the AdobeStockClient Test section. And in this class, I will call represented classes in this PR. Am i right?

**Comment by coderimus** [comment]
@sivaschenko I tested approach I provided at the previous message and can say that it is not good enough. Ignore my previous suggestions.

I made an investigation and I suggest to make a stub, not on the client class bu on the `Magento\AdobeStockClient\ModelConnectionWrapper`. This propose looks like the right solution because we stub the service which communicates with the Adobe Stock SDK and our implementation.

The one last step before I will proceed is to discuss how to provide this stub functionality. Maybe we can play with the autoload and make this module accessible for the Magento module installation? I mean we can try to add an autoload rule for this stub class (or make a module) and place it under the Test\_files forlder or else in the AdobeStockClient namspace? What are your thoughts about this?

**Comment by sivaschenko** [comment]
@coderimus I am ok with replacing the `ConnectionWrapper` by test module. I don't think any modifications to autoload are needed. Please use DI preference functionality to rewrite the ConnectionWrapper.

**Comment by sivaschenko** [comment]
@coderimus any updates on this pr, are you still planning to work on it?

**Comment by coderimus** [comment]
@sivaschenko yes, this PR is in progress on my board. Next update will be on the 14th of January.

**Comment by coderimus** [comment]
@sivaschenko I provided the initial commit to the `AdobeStockClientStub` module. This module represents the stub which should provide desired functionality and simplicity in use.
My vision of its use is next: reference it in the `di.xml` instead fo the `Magento\AdobeStockClient\Model\Client` implementation. The stub Client service implements  the `Magento\AdobeStockClientApi\Api\ClientInterface` also.

Every method represented as a separate pure class with a connected template where the stub data creates. This makes possible future customisations for the stub logic more isolated and simple for use.

Currently, I covered the next methods with the stub data: `search`, ` getQuota`, `getLicenseConfirmation`.  Other four methods will be included in the next several days. But feel free to check existed implementation :)

File: dev/tests/acceptance/.credentials [hunk 1]
```diff
@@ -1,4 +1,4 @@
-magento/ADOBE_STOCK_API_KEY=%ADOBE_STOCK_API_KEY%
-magento/ADOBE_STOCK_PRIVATE_KEY=%ADOBE_STOCK_PRIVATE_KEY%
-magento/ADOBE_STOCK_USER_EMAIL=%ADOBE_STOCK_USER_EMAIL%
-magento/ADOBE_STOCK_USER_PASSWORD=%ADOBE_STOCK_USER_PASSWORD%
+magento/ADOBE_STOCK_API_KEY=adove_stock_api_key
```
Comment (Nazar65) [inline dev/tests/acceptance/.credentials:1]:
```suggestion
magento/ADOBE_STOCK_API_KEY=adobe_stock_api_key
```

File: AdobeStockClientStub/Model/ConnectionWrapper.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+use Magento\Framework\Exception\AuthorizationException;
+use Magento\Framework\Exception\IntegrationException;
+
+/**
+ * Emulate ConnectionWrapper functionality
+ */
+class ConnectionWrapper
+{
+    private const INVALID_API_KEY = 'wrong-api-key';
+    private const VALID_API_KEY = 'dove_stock_api_key';
```
Comment (Nazar65) [inline AdobeStockClientStub/Model/ConnectionWrapper.php:24]:
```suggestion
    private const VALID_API_KEY = 'adobe_stock_api_key';
```

File: dev/tests/acceptance/.credentials (hunk 1 above)
Comment (coderimus) [reply dev/tests/acceptance/.credentials:1]: @Nazar65 thank you! Will change.

File: AdobeStockClientStub/Model/ConnectionWrapper.php (hunk 2 above)
Comment (coderimus) [reply AdobeStockClientStub/Model/ConnectionWrapper.php:24]: Thank you. Will change.

**Comment by hws47a** [comment]
Hi @coderimus, in the description it says "Work is in progress.". Please update the description if you want it to be reviewed.

**Comment by coderimus** [comment]
I close this PR and will open the new one. This one mostly a proof of concept.

**Pull request closed without merging**